### PR TITLE
Add encoder_cls argument to CacheFileHandler

### PR DESCRIPTION
**Pull request by shawncruz**
Resolves https://github.com/spotipy-dev/spotipy/issues/940

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -39,6 +39,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 - Unit tests for new releases, passing limit parameter with minimum and maximum values of 1 and 50
 - Unit tests for categories, omitting country code to test global releases
 - Added `CODE_OF_CONDUCT.md`
+- Added optional `encoder_cls` argument to `CacheFileHandler`, which overwrite default encoder for token before writing to disk
```
Comment (stephanebruckert) [inline CHANGELOG.md:42]: please add it in the unreleased section

**Review by stephanebruckert — COMMENTED** [review]
looks good, thanks! just one comment on the changelog

File: CHANGELOG.md (hunk 1 above)
Comment (shawncruz) [reply CHANGELOG.md]: Updated!

**Review by stephanebruckert — APPROVED** [review]

**Comment by stephanebruckert** [comment]
There likely won't be a release for it so soon but feel free to use @master or @b3f308d in your code for now :-)

**Pull request merged**